### ISSUE TITLE
Use `SigningKey`/`VerifyingKey` naming convention?

### ISSUE THREAD

**Issue by tarcieri** [dalek-cryptography/ed25519-dalek#225]
In the @RustCrypto project we have adopted a convention of naming signature keys `SigningKey` and `VerifyingKey` which describes keys by their *roles* in a signature system.

The basic pitch is that such names are clearer and more intuitive to users who don't necessarily have a background in cryptography.

Would it make sense to adopt similar names in this crate?

**Comment by rozbb** [comment]
This sounds like a reasonable thing to do for 2.0

**Comment by burdges** [comment]
You're not bothered and/or like the overlap with trusted setups?

**Comment by burdges** [comment]
Yeah.  I miss-phrased the question.  I meant: Did you think about it?  I've no opinion really.

**Comment by tarcieri** [comment]
They serve similar roles, but typical signature algorithms are proving knowledge of the signature key

**Comment by tarcieri** [comment]
A semi-related question (which perhaps deserves its own issue) is if the `SecretKey` and `Keypair` types should be combined into a single type, e.g. `SigningKey`.

Especially after #205, there is little value in keeping them separate, as the related API surface was exposed.

I would suggest adopting the RFC8032 definition of a secret key being 32-bytes. IMO the 64-byte "keypair" types (which contain the 32-byte secret key + 32-byte public key which can be derived from the secret key) is something of an accidental relic of the `ref10` implementation of Ed25519 from SUPERCOP being somewhat haphazardly repurposed into production code, i.e. the C struct representing a keypair being effectively transmuted into bytes and considered a key.

The recent public key oracle attack shows that not only is using that 64-bit serialization as a wire format needlessly wasteful, since the public key can be derived from the secret key, but it also introduces potential misuse. If we were to keep it around (i.e. the current `Keypair::from_bytes`), e.g. as `SigningKey::from_keypair_bytes` or so forth, I would suggest deprecating it and trying to move people away from that format towards RFC8032 secret keys.

**Comment by rozbb** [comment]
I'm pro removing `Keypair`. Whenever possible, I'd like to avoid defining semantics for data structures whose functions overlap. How will the deprecation interact with #224?

**Comment by tarcieri** [comment]
It'd probably be good to land #224 first before making these changes.

I think really it'd be folding the existing `Keypair` type and `SecretKey` type into a single `SigningKey` type. The signing functionality is impl'd on `Keypair` because it needs access to the public key to deterministically compute the unique per-message pseudorandom scalar used to compute the signature.

**Comment by pinkforest** [comment]
The PR got merged
- #242
